### ISSUE TITLE
[Linter] Invalid Linter Result received

### ISSUE THREAD

**Issue by xumukns** [tvincent056/linter-gcc2#3]
## Checklist
* [ ] Check that you have the GNU C Compiler installed by typing ```gcc --version``` into a terminal.
MacOS
/usr/local/gcc-8.2/bin/gcc-8.2 -v
Es werden eingebaute Spezifikationen verwendet.
COLLECT_GCC=/usr/local/gcc-8.2/bin/gcc-8.2
COLLECT_LTO_WRAPPER=/usr/local/gcc-8.2/libexec/gcc/x86_64-apple-darwin18.2.0/8.2.0/lto-wrapper
Ziel: x86_64-apple-darwin18.2.0
Konfiguriert mit: ../configure --prefix=/usr/local/gcc-8.2 --enable-checking=release --with-gmp=/usr/local/gcc-8.2 --with-mpfr=/usr/local/gcc-8.2 --with-mpc=/usr/local/gcc-8.2 --enable-languages=c,c++,fortran --with-isl=/usr/local/gcc-8.2 --program-suffix=-8.2
Thread-Modell: posix
gcc-Version 8.2.0 (GCC) 

* [yes ] If you don't have the ```linter``` package or ```atom-ide-diagnostics``` package installed, either update to the latest version of ```linter-gcc2``` or install linter or atom-ide-diagnostics directly with ```apm install linter``` or ```apm install atom-ide-diagnostics```.
* [yes ] Ensure that you are using a valid grammar - either C/C++/C++14
* [yes ] Ensure that you have read all the relevant information in the [wiki](https://github.com/hebaishi/linter-gcc/wiki), particularly the [Common Issues](https://github.com/hebaishi/linter-gcc/wiki/Common-Issues) page.

## Issue (including debugging messages):
[Linter] Invalid Linter Result received
These issues were encountered while processing messages from a linter named 'GCC'
  • Message.location must be valid

**Comment by tvincent056** [comment]
@xumukns Could you provide a bit more detail? What steps did you take to get this error? Did you have linter-on-the-fly enabled or disabled?

**Comment by xumukns** [comment]
@tvincent056 Hi! I have installed gcc7. It works fine. At first, I have installed the linther-gcc it gives this error. After that, I have found the linther-gcc2 and installed it after the deinstallation the first one. The error message is the same. The message comes on the both linter-on-the-fly  - enabled and disabled.
If the linter-on-the-fly disabled, the message comes after the activation of the linther-check -  for example by saving of the file.

**Comment by tvincent056** [comment]
@xumukns I have made a few updates since you opened this issue. Have you tried again lately? Are you still having problems?

**Comment by xumukns** [comment]
@tvincent056 Thank you! The mentioned error message is gone. Now I get every time this error message:
Severity 	Provider 	Description	Line 
Error	GCC	clang -cc1as: error: unable to open output file '8c2de465-09ad-46eb-bd3a-010f3251c9f3.o': 'Permission denied' 1:1

**Comment by tvincent056** [comment]
@xumukns Based on your original error message, it looks like you are using this on Mac OS. In your latest comment, it looks like the error message mentions clang (which is an entirely different compiler). I have not tried atom + linter-gcc2 on Mac OS, but there is documentation in the README for this plugin (carried over from the original linter-gcc) which indicates that you need apply a workaround to get the plugin to use gcc rather than clang. Have you tried the workaround?

**Comment by xumukns** [comment]
Yes, I am using it MacOS. I have installed gcc 8.3 on it and it works fine. All environment variables (PATH, LD_LIBRARY_PATH, …etc.) are set correctly. In the Plugin configuration I have set full path to gcc-8.3 and I see the plugin uses it.


Because if I change the GCC exec.path „…/gcc-8.3“ to something not existing like „…/gcc-8.4“ plugin pops a lot of error messages like „executable is not found" and so on....
In the mentioned configuration I get only one error message from clang. That means the Linter-Plugin uses gcc (linked to clang) command independent from the configuration setting. Or I see it wrong?

**Comment by xumukns** [comment]
Configuration snapshot:
![bildschirmfoto 2019-02-25 um 15 50 16](https://user-images.githubusercontent.com/26707530/53346184-d0673800-3916-11e9-9d83-949afdfa9785.png)

**Comment by xumukns** [comment]
I am sorry. Should I move the new issue in my case to another thread?

**Comment by tvincent056** [comment]
@xumukns It looks like the underlying compiler is trying to save an object file in a place that you don't have write permissions. We don't need the object files and you can disable their output in gcc using -fsyntax-only. Have you tried adding -fsyntax-only to your C Flags or C++ Flags (whichever is appropriate)?

**Comment by xumukns** [comment]
@tvincent056 Yes, "-fsyntax-only" works for me. Thank you. Unfortunately, I see another problem now: It reports all errors always at line 1:1 independent from the real position of the wrong string. The gcc message in the description contains the real position of the syntax or programming error, but the plugin does not show it. It shows only a red point in line 1 position 1.

**Comment by tvincent056** [comment]
That seems like a parsing error (plugin couldn't read position correctly from gcc output). Could you send an example?

**Comment by xumukns** [comment]
As I said it is gcc 8.3. The programm:
/////////////////////////////////////
#include <stdlib.h>
int main(int argc, const char *argv[]){
  int f;
  float ff;
  ff = i;

  return 0;
}
//////////////////////////////////
The Linter error message:
![bildschirmfoto 2019-02-25 um 17 01 11](https://user-images.githubusercontent.com/26707530/53350270-0dcfc380-391f-11e9-9c58-4e4ae596a448.png)
